### ISSUE TITLE
Add http endpoint for making transfers

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#440]
## Problem Definition

The REST API server does not have a endpoint for making transfers.

## Solution

Add an endpoint for transfers, without exposing the details of different transfer type, e.g. direct vs. mediated.

Enpoint:

`POST /api/1/transfer/:transfer_token/:transfer_target/:transfer_amount`

Return codes:

**200** - When the transfer was sucessfully executed (the secret was revealed, there is no need to block until the balance proofs are exchanged).
**400** - When a timeout happened, this timeout doesn't need to be the lock's expiration but only the raiden protocol timeout, since this is the node starting the request and it controls the secret.

**Comment by LefterisJP** [comment]
I am working on this one now.

@hackaugusto I will change the spec a bit to be consistent with the rest of the API.

Proposal:

```POST /api/<version>/transfers/<token_address>/<target_address>```

with a json payload of the transfer details which for now would only contain:

```json
{
    "amount": 1042,
    "identifier": 3
}
```

Additionally I would like to include one more return code. 

**409** - If there is not enough balance in the initiator to even begin the transfer (or some other such error)

Finally, but this is out of the scope of this issue (should be part of https://github.com/raiden-network/raiden/issues/466) doing a `GET` on `/api/<version>/transfers/<token_address>/<target_address>` or on `/api/<version>/transfers/<token_address>` it should return all known transfers that occured to either that target address for a token or (second URL) all transfers for a token.

Please provide feedback so that implementation can proceed.

**Comment by hackaugusto** [comment]
About the URL+JSON, definitely my proposal was wrong, completely agree that we need to change to plural and remove the `transfer_amount` from the URL in favor of JSON. One suggestion, perhaps we should use the PUT verb instead of POST with the identifier of the transfer in the URL?

About the HTTP ERROR codes:

[408 - REQUEST TIMEOUT](https://httpstatuses.com/408): I had gave this one a bit of thought before, the main problem is that 4xx are _client_ errors, and 408 is used when the _client_ didn't send the data to the server in due time, so the server closes the connection because it's not willing to wait anymore. That's not really what we want, we want to inform the client that a downstream timeout happened. We could overload this code for our purposes, but I'm not 100% sure if that is a good idea. (I have looked at most of the HTTP ERROR codes and I didn't find any that matches our needs exactly. Actually for this case a 5xx 
error is probably better).

[409 - CONFLICT](https://httpstatuses.com/409): Sounds good to use this one.
